packages/headless/src/features/facets/category-facet-set/category-facet-set-slice.test.ts

```diff
@@ -786,17 +788,43 @@ describe('category facet slice', () => {
       state[facetId] = buildMockCategoryFacetSlice({request});
 
       const facet = buildMockCategoryFacetResponse({facetId});
-      const action = buildExecuteSearchAction([facet]);
+      const action = searchBuilder([facet]);
       const finalState = categoryFacetSetReducer(state, action);
 
       expect(finalState[facetId]?.request.preventAutoSelect).toBe(false);
     });
 
     it('when the facet response #id does not exist in state, it does not throw', () => {
       const facet = buildMockCategoryFacetResponse({facetId});
-      const action = buildExecuteSearchAction([facet]);
+      const action = searchBuilder([facet]);
 
       expect(() => categoryFacetSetReducer(state, action)).not.toThrow();
     });
+  }
+
+  describe('#executeSearch.fulfilled', () => {
+    function buildExecuteSearchAction(facets: FacetResponse[]) {
+      const search = buildMockSearch();
+      search.response.facets = facets;
+
+      return executeSearch.fulfilled(search, '', logSearchEvent({evt: 'foo'}));
+    }
+
+    testFulfilledSearchRequest(buildExecuteSearchAction);
+  });
+
+  describe('#fetchFacetValues.fulfilled', () => {
+    function buildFetchFacetValuesAction(facets: FacetResponse[]) {
+      const search = buildMockSearch();
+      search.response.facets = facets;
+
+      return fetchFacetValues.fulfilled(
+        search,
+        '',
+        logSearchEvent({evt: 'foo'})
+      );
+    }
+
+    testFulfilledSearchRequest(buildFetchFacetValuesAction);
```

Comment:
I think a single test checking the happy path of `fetchFacetValues` would be enough here.

The current approach has the plus of being "extra safe", but running the block of tests twice just means tests will fail in unison on any source change? With a single new test, we'd have the same coverage, but have fewer tests at run-time and avoid duplicate failures.